applications/postgres.md

```diff
@@ -1,13 +1,222 @@
+
+*	3 node Portworx storage Cluster
+
+### Step 1 - Create PostgreSQL Portworx StorageClass(sc)
+Create a file name called `px-postgres-sc.yaml` for Portworx StorageClass(sc) for PostgreSQL.
+
+```
+
+##### Portworx storage class
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+    name: px-postgres-sc
+provisioner: kubernetes.io/portworx-volume
+parameters:
+   repl: "2"
+
+```
+`sudo kubectl apply -f px-postgres-sc.yaml`
+
+### Step 2 - Create PostgreSQL Portworx PersistentVolumeClaim(pvc)
+Create a file name called `px-postgres-vol.yaml` for Portworx PersistentVolumeClaim(pvc) for PostgreSQL.
+
+```
+
+##### Portworx persistent volume claim
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+   name: postgres-data
+   annotations:
+     volume.beta.kubernetes.io/storage-class: px-postgres-sc
+spec:
+   accessModes:
+     - ReadWriteOnce
+   resources:
+     requests:
+       storage: 1Gi
+
+```
+`sudo kubectl apply -f px-postgres-vol.yaml`
+
+### Step 3 - Deploy PostgreSQL using Kubernetes Deployment
+ 
+Deploying PostgreSQL on Kubernetes have following prerequisites. 
+Create a file name `px-postgres-app.yaml` for PostgreSQL. 
+ 
+We need to Define the Environment Variables for PostgreSQL
+
+1. POSTGRES_USER (Super Username for PostgreSQL)
+
+2. POSTGRES_PASSWORD (Super User password for PostgreSQL)
+
+3. PGDATA (Data Directory for PostgreSQL Database)
+
+```
+
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: postgres
+spec:
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: postgres
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: px/running
+                operator: NotIn
+                values:
+                - "false"
+              - key: px/enabled
+                operator: NotIn
+                values:
+                - "false"
+      containers:
+      - name: postgres
+        image: postgres:9.5
+        imagePullPolicy: "IfNotPresent"
+        ports:
+        - containerPort: 5432
+        env:
+        - name: POSTGRES_USER
+          value: pgbench
+        - name: POSTGRES_PASSWORD
+          value: superpostgres
+        - name: PGBENCH_PASSWORD
+          value: superpostgres
+        - name: PGDATA
+          value: /var/lib/postgresql/data/pgdata
+        volumeMounts:
+        - mountPath: /var/lib/postgresql/data
+          name: postgredb
+      volumes:
+      - name: postgredb
+        persistentVolumeClaim:
+          claimName: postgres-data
+
+``` 
+
+ Now let's deploy PostgreSQL using following commands:
+
+`sudo kubectl apply -f px-postgres-app.yaml`
+
+### Validate StorageClass(sc), PersistentVolumeClaim(pvc) and PostgreSQL Deployment.
+```
+ubuntu@node1:~$ sudo kubectl get sc
```

Comment:
remove `sudo`. It is a result of how kubectl is setup on a particular system and doesn't need to bubble up on generic steps here.